### PR TITLE
Separate tell api into complete api and fail api

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

To address https://github.com/optuna/optuna-dashboard/pull/355#discussion_r1083662852

## What does this implement/fix? Explain your changes.

Separate tell api into complete api and fail api

### CODE REVIEW

File: optuna_dashboard/ts/action.ts [hunk 1]
```diff
@@ -444,16 +444,44 @@ export const actionCreator = () => {
...
-    state: TrialStateFinished,
-    values?: number[]
+    values: number[]
   ): void => {
-    const message =
-      values === undefined
-        ? `id=${trialId}, state=${state}`
-        : `id=${trialId}, state=${state}, values=${values}`
+    const state = "Complete"
+    const message = `id=${trialId}, state=${state}, values=${values}`
```
Comment (c-bata) [inline optuna_dashboard/ts/action.ts:484]: It looks that state is undefined.

```suggestion
    const message = `id=${trialId}, state=Complete, values=${values}`
```

Comment (c-bata) [reply optuna_dashboard/ts/action.ts:484]: Or might be good to update L497 like:

```
enqueueSnackbar(`Successfully make trial completed (${message})`, {
    variant: "success",
})
```

File: optuna_dashboard/ts/action.ts [hunk 2]
```diff
@@ -444,16 +444,44 @@ export const actionCreator = () => {
       })
   }
 
-  const tellTrial = (
+  const makeTrialFail = (studyId: number, trialId: number): void => {
+    const state = "Fail"
+    const message = `id=${trialId}, state=${state}`
```
Comment (c-bata) [inline optuna_dashboard/ts/action.ts:449]: I think it doesn't need to declare the `state` variable.

```suggestion
    const message = `id=${trialId}, state=Fail`
```

**Comment by keisuke-umezawa** [comment]
@c-bata Thank you for reviewing! I followed your comment.

**Pull request merged**